### PR TITLE
[VL] Fix spark34 group-by.sql(.out) in GlutenSQLQueryTestSuite

### PR DESCRIPTION
**Pull request by liujiayi771**
## What changes were proposed in this pull request?

#4655 enabled `group-by.sql` test case in `GlutenSQLQueryTestSuite`, but it copied the spark33's group-by.sql(.out) to spark34 directory. These files have some modifications in Spark 3.4, and we should overwrite the files in Spark 3.4.

## How was this patch tested?

Exists CI.

### CODE REVIEW

**Comment by liujiayi771** [comment]
cc @rui-mo @ulysses-you , thanks.

File: gluten-ut/spark33/src/test/scala/io/glutenproject/utils/velox/VeloxSQLQueryTestSettings.scala [hunk 1]
```diff
@@ -234,7 +234,7 @@ object VeloxSQLQueryTestSettings extends SQLQueryTestSettings {
   private val OVERWRITE_SQL_QUERY_LIST: Set[String] = Set(
     // Velox corr has better computation logic but it fails Spark's precision check.
     // Remove -- SPARK-24369 multiple distinct aggregations having the same argument set,
-    // -- SPARK-37613: Support ANSI Aggregate Function: regr_r2
+    //        -- SPARK-37613: Support ANSI Aggregate Function: regr_r2
```
Comment (ulysses-you) [inline gluten-ut/spark33/src/test/scala/io/glutenproject/utils/velox/VeloxSQLQueryTestSettings.scala:237]: do you want to remove this comment ?

Comment (liujiayi771) [reply gluten-ut/spark33/src/test/scala/io/glutenproject/utils/velox/VeloxSQLQueryTestSettings.scala:237]: I just want to align these two lines.
```
// Velox corr has better computation logic but it fails Spark's precision check.
// Remove -- SPARK-24369 multiple distinct aggregations having the same argument set,
//        -- SPARK-37613: Support ANSI Aggregate Function: regr_r2
"group-by.sql",
```

**Review by ulysses-you — APPROVED** [review]
lgtm, thank you

**Pull request merged**